kornia/geometry/epipolar/projection.py

```diff
@@ -108,6 +108,43 @@ def projection_from_KRt(K: torch.Tensor, R: torch.Tensor, t: torch.Tensor) -> to
     return K @ Rt
 
 
+def KRt_from_projection(P: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
+    r"""This function decomposes the Projection matrix into Camera-Matrix, Rotation Matrix and Translation vector.
+
+    Args:
+        P (torch.Tensor): the projection matrix with shape :math:`(B, 3, 4)`.
+
+    Returns:
+        Tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
+        - The Camera matrix with shape :math:`(B, 3, 3)`.
+        - The Rotation matrix with shape :math:`(B, 3, 3)`.
+        - The Translation vector with shape :math:`(B, 3)`.
+
+    """
+
+    assert P.shape[-2:] == (3, 4), "P must be of shape [B, 3, 4]"
+
+    submat_3x3 = P[:, 0:3, 0:3]
+    last_column = P[:, 0:3, 3].unsqueeze(-1)
+    t = torch.matmul(-torch.inverse(submat_3x3), last_column).squeeze(-1)
+
+    # Trick to turn QR-decomposition into RQ-decomposition
+    reverse = torch.Tensor([[0, 0, 1], [0, 1, 0], [1, 0, 0]], device=P.device, dtype=P.dtype).unsqueeze(0)
```

Comment:
![image](https://user-images.githubusercontent.com/5157099/91815800-1ea87600-ec35-11ea-9b7e-7e09e886b0fb.png)

looking at the above picture from the links you shared, I think we could de compose this function other small utilities that could be very useful. E.g `flipud` which is already a numpy function and very easy to implement. See: https://numpy.org/doc/stable/reference/generated/numpy.flipud.html
```python
A = np.random.randn(2,3,5)
np.all(np.flipud(A) == A[::-1,...])
True
```

second, why not having as separate function also `kornia.rq` which implements `K.flipud` and `torch.qr`. This is a suggestion that can be included in a separated pull request

Reply:
Yes, I will break the whole function into smaller utilities, but in the next PR.